11-Backbonejs/Backbone-Views.md

```diff
@@ -0,0 +1,321 @@
+
+### Handling Creating New Tasks
+
+Just like we added event handlers in the `TaskView` to handle button clicks we can add an event handler to create a new task and a method to read from the new task form, and add them to the collection.
+
+
+```javascript
+// views/task_list_view.js
+}, // end of render
+
+events: {
+    'click #add-task': "addTask"
+  },
+  readNewTaskForm: function() {
+    // Get the values from the fields
+    var title = this.$('#title').val();
+    this.$('#title').val('');
+    var description = this.$('#description').val();
+    this.$('#description').val('');
+    var completed = this.$('#completed-checkbox').is(":checked");
+    this.$('#completed-checkbox').prop('checked', false);
+
+    return {
+      title: title,
+      description: description,
+      completed: completed
+    };
+  },
+  addTask: function(e) {
+    var taskData = this.readNewTaskForm();
+    var task = new Task(taskData);
+    this.model.add(task);
+  }
+```
+
+Again this looks very much like what we originally wrote in `app.js`  We did change the code to use `this.$` instead of direct jQuery and called `readNewTaskForm()` as an instance method with `this.`.  
+
+**Note you will need to delete the original event handlers in `app.js`.**
+
+**Gah!** The Add button ALSO doesn't work anymore!
+
+Good point, the delete and add buttons are actually working, and tasks are getting added and removed from the collection, but the problem is we're not re-rending the list.
+
+## Adding a Backbone Event Listener
+
+If we add an event listener to the collection we can call render when the collection is updated.  
+
+Similar to `on` we can use a method called `listenTo` to add an event listener for our collection.  We will do so in the `initialize` method.
+
+```javascript
+  initialize: function(params) {
+    this.template = params.template;
+
+    this.listenTo(this.model, "update", this.render);
+  },
+```
+
+## Last bit, adjusting styles
+
+You may have noticed that the styling is a bit broken.  That's because the `li` tag we set in the `TaskView`  doesn't have any classes added to it for styling.  
+
+**NOTE** this is for `task_view.js` not `task_list_view.js`.  
+
+
+```javascript
+// src/views/task_view.js
+
+  initialize: function(params) {
+    this.template = params.template;
+    
+    this.listenTo(this.model, "update", this.render);
+    
+	 // Add classes for styling
+    this.$el.addClass("task-item");
+    this.$el.addClass("column");
+    this.$el.addClass("column-block");
+  },
+```
+
+We also need to make one adjustment by removing the `li` tag from the HTML template.
+
+## Summary
+
+So now we have a solution which displays our list of tasks, lets us toggle them complete or incomplete, delete and create tasks.  The view acts as a coordinator handling events and coordinating the response with the template & the model.  
+
+
+## Optimizations & Extensions
```

Comment:
It feels like it is too soon after the initial introduction to highlight optimizing the code - unless this is an expectation that we'd expect the students to do in their own assignments